contracts/Distributor.sol

```diff
@@ -0,0 +1,250 @@
+        require(accounts_.length == shares_.length, "Distributor: Mismatch of addresses and shares.");
+        for(uint256 i = 0; i < accounts_.length; i++ ){
+            _registerPayeesForETH(accounts_[i], shares_[i]);
+        }
+    }
+
+     /*///////////////////////////////////////////////////////////////
+                            External/Public Functions
+    //////////////////////////////////////////////////////////////*/
+
+    // @Desc: To batch register accounts with a specified share for a new ERC20 Token
+    function registerPayeesForToken(address[] memory accounts_, IERC20 token_, uint256[] memory shares_) external onlyOwner {
+        require(accounts_.length == shares_.length, "Distributor: Mismatch of addresses and shares.");
+        for(uint256 i = 0; i < accounts_.length; i++ ){
+            _registerPayeesForTokens(accounts_[i], token_, shares_[i]);
+        }
+    }
+    
+    // @Desc: Payout/Claim eligible amount of ether from contract based on shares
+    function payoutETH(address payable account_) public {
+        address payable account = account_;
+        require(_sharesForETH[account_] > 0, "Distributor: Not registered to claim ETH.");
+
+        uint256 ethClaimable = claimableETH(account);
+
+        //Send ether to receipient via a low-level call function
+        require(address(this).balance >= ethClaimable, "Distributor: Insufficient ETH balance in contract.");
+        
+        _totalETHDistributed += ethClaimable;
+        _distributedETH[account] = ethClaimable;
+        (bool success,) = account.call{value: ethClaimable}("");
+        require(success, "Distributor: Unable to receive ETH from contract.");
+
+        emit ETHFundClaimed(account, ethClaimable);
+    }
+
+    function payoutToken(address account_, IERC20 token_) public {
+        address account = account_;
+        IERC20 token = token_;
+        require(_sharesForTokens[token][account] > 0, "Distributor: Not registered to claim ERC20 token.");
+
+        uint256 tokenAmountClaimable = claimableToken(account, token);
+
+        // Transfer ERC20 via transfer function from SafeERC20
+        require(token.balanceOf(address(this)) >= tokenAmountClaimable, "Distributor: Insufficient ERC20 token balance.");
+        
+        _totalTokenDistributed[token] += tokenAmountClaimable;
+        _distributedTokens[token][account] += tokenAmountClaimable;
+        SafeERC20.safeTransfer(token, account, tokenAmountClaimable);
+
+        emit TokenFundClaimed(account, token, tokenAmountClaimable);
+    }
+
+    // @Desc: Convenient One-Time Function call for owner to distribute ether from contract to all registered payees
+    function distributeDemAllETH() external onlyOwner {
+        uint totalPayees = numRegisteredForETH();
+        for(uint256 i = 0; i < totalPayees;){
+            if(_sharesForETH[_payeesForETH[i]] > 0) {
+                payoutETH(payable(_payeesForETH[i]));
+            }
+            unchecked {i++; }
+        }
+    }
```

Comment:
this will not revert if there is a payee does not have ETH allocated, unlike your below function for distributeDemAllToken will revert if this happens => inconsistent

Reply:
Oh yes sorry I overlooked this part, I could remove that line of code too because totalPayee refers to the actual number of accounts registered with shares for ETH, will amend it accordingly thanks!

Reply:
May be that they will not able to claim exactly amount they want ?

Reply:
They will be able to claim the proportion of funds deposited inside the contract based on their shares relative to the total shares, regardless of how many times it is being topped up